### ISSUE TITLE
[Proposal] Improving accessibility

### ISSUE THREAD

**Issue by nmuranovas** [gorhom/react-native-bottom-sheet#548]
## Ask your Question

Hey, thank you for creating such a promising library!
Some context - my team works for a large company and this library has caught our eyes as it has almost everything that we need. Since accessibility is one of our top priorities - we wanted to ask if we could contribute by improving the accessibility of the bottom sheet?

## Proposal

Here is a list of changes we are willing to do. Please let us know if the proposal interests you.

[BottomSheetBackdrop] make the following props overridable:
* accessible;
* accessibilityRole;
* accessibilityLabel;
* accessibilityHint.

[BottomSheet]
* make the following props overridable:
    * accessible;
    * accessibilityRole;
    * accessibilityLabel;
* add the following props:
    * announcePositionChange?: boolean:
        * if disabled - does not announce the position changes on screen reader;
        * by default is enabled (true);
    * onPositionChangeAnnounce?: (positionInScreen) => string;
        * allows the customisation of position change announcement for screen reader;
        * takes a callback that returns a string that will be read out by the screen reader.

[BottomSheetHandleContainer] move accessibility props to [BottomSheetHandle] and make the following props overridable:
* accessible;
* accessibilityRole;
* accessibilityLabel;
* accessibilityHint.

When setting accessibility to false on any of the components - screen reader should not detect the elements.
Fix bug where if modal is dismissed it announces that it has snapped to 10%.

**Comment by gorhom** [comment]
Thanks @nmuranovas for this details enhancement/suggestion , i am very happy with this , i will check your PR asap 👍

**Comment by nmuranovas** [comment]
Bump - still relevant. The PR is waiting for review

**Comment by gorhom** [comment]
thanks @nmuranovas, now it is available for `v2` on `2.4.0` release. I will try to work on replicate the implementation for `v4` later, so i will keep this ticket as a reminder

**Comment by nmuranovas** [comment]
@gorhom I can take up the task of implementing this in v4. Would that be ok?

**Comment by gorhom** [comment]
@nmuranovas that would be great <3

**Comment by jsamr** [comment]
@nmuranovas Are you still willing to port this?

**Comment by nmuranovas** [comment]
@jsamr Yes, I was just working on it. Although, it seems like v4 does not have the same accessibility functionality as v2 so I will take some extra time to code and test out on both ios and android :)

**Comment by YanislavSpaceIQ** [comment]
> @jsamr Yes, I was just working on it. Although, it seems like v4 does not have the same accessibility functionality as v2 so I will take some extra time to code and test out on both ios and android :)

Any news ?  @gorhom @nmuranovas

**Comment by clement-livdeo** [comment]
> @jsamr Yes, I was just working on it. Although, it seems like v4 does not have the same accessibility functionality as v2 so I will take some extra time to code and test out on both ios and android :)

Is there any news about a11y in v4 ? @gorhom @nmuranovas

**Comment by hongbinhuang** [comment]
> Is there any news about a11y in v4 ? @gorhom @nmuranovas

@gorhom @nmuranovas Same question here, appreciate it if you can provide some update on the v4 improvement, thanks!
